Sources/Sentry/SentryTracer.m

```diff
@@ -816,6 +860,74 @@ + (nullable SentryTracer *)getTracer:(id<SentrySpan>)span
     return nil;
 }
 
+#pragma mark - Extensions
```

Comment:
I disagree. This IMO is the simplest solution. 
If we dont use SentryTracerExtension we would need to distribute more responsibility to SentryTracer and SentryUIViewControllerPerformanceTracker.

Reply:
What other solutions did you consider such that you arrived at this one being the simplest? The only class that will conform to this protocol so far is the TTD tracker. What other classes do you plan to conform to it? 

Couldn't SentryTracer directly instantiate those classes, store the references as properties, and interact directly with them, without the extra machinery? Then it would be a simple composition hierarchy. There is some complexity involved with the type-erasure (`id<SentryTracerExtension>`), storing them in an array that has to be synchronized and copied, etc.